File: api-java/demo/src/main/java/com/example/demo/model/Task.java
package com.example.demo.model;

import java.time.LocalDateTime;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Task {

    private Long id;
    private String title;
    private String description;
    private TaskStatus status;
    private LocalDateTime createAt;
    private LocalDateTime updatedAt;

    public Task(){
    }

    public Task(Long id, String title, String description, TaskStatus status){
        this.id = id;
        this.title = title;
        this.description = description;
        this.status = status;
        this.createAt = LocalDateTime.now();
    }

}

File: frontend-react/todo-react/src/types/Task.ts
export interface Task{
    id: number;
    title: string;
    description: string;
    status: TaskStatus;
    createAt?: Date;
    updatedAt?: Date;
}

export interface FilterSortTask{
    title?: string;
    description?: string;
    titleAndDescription?: string;
    status?: TaskStatus[];
    createdFrom?: Date;
    createdTo?: Date;
    updatedFrom?: Date;
    updatedTo?: Date;
    page?: number;
    size?: number;
    sortBy?: string;
    direction?: string;
}

export interface CreateUpdateTask{
    title?: string;
    description?: string;
    status?: TaskStatus;
}

export enum TaskStatus {
    TODO = 'TODO',
    IN_PROGRESS = 'IN_PROGRESS',
    COMPLETED = 'COMPLETED',
    CANCELLED = 'CANCELLED'
}